### ISSUE TITLE
Reorganize install instructions as interface-specific checklist

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/rstan#324]
#### Summary:

Make the installation instructions for RStan
- easier to follow for users, and 
- easier for us to diagnose when users have problems.
#### Description:
1.  Organize instructions by interface.  Have a separate page of instructions for each platform (Windows, Mac OSX, Linux).  These can even be broken down by version of OS if they're different in more  than one place.  It's the combinatorics of the instructions that are daunting.
2. Organize instructions as step-by-step checklist.
3.  Ideally, for each step, provide a way to verify that it was done correctly.
4.  Even better, provide troubleshooting advice related to checklist items to help users self-diagnose problems and allow us to see at exactly which step things went wrong for user installs.

**Comment by bob-carpenter** [comment]
I don't think I can do this myself, but I'd be glad to act as a guinea pig and try them out if I can get my hands on machines that don't already have RStan and required pre-reqs installed properly.

**Comment by bgoodri** [comment]
Aside from the toolchain, the installation instructions aren't really platform specific. The configuration is somewhat platform-specific, but it is really more compiler specific. As in, this is what you need for clang++ but it is essentially the same whether you are using clang++ on a Mac or Linux.

**Comment by bob-carpenter** [comment]
I know it's redundant, but our users are having a very hard
time following the intricate, conditional instructions.  I
fear potential users are just bouncing and not using Stan because
they can't get RStan installed.

I think it would be much much easier if there were 
platform-specific instructions for installing, and
if platform is OS + compiler, then I think we should do
that.

The instructions currently contain things like:

```
In addition, on OS X only you should (unless you do not have clang++ installed)
execute in R...
```

and

```
Starting with R version 3.3.x, it is possible to download Rtools for Windows
that uses g++ 4.9.x, which supports the C++11 standard. Using the C++11
standard is not currently by supported by Stan for versions of g++ up to
and including 4.6 but is believed to work for later versions of g++ and
any recent version of clang++.
```

I think these instructions are a large part of the problem.
I believe many our users don't know how to figure out which compiler
they have installed, much less follow these kinds of instructions.

**Comment by jgabry** [comment]
Yeah I think most if not all of the info is already there but I agree that
some of it is hard to follow unless you're familiar with it already. I like
the idea of splitting it up by platform even if a lot of it gets copied.
We'll have to make sure that when we update the common content in one of
them we update all of them, but that's seems preferable to how it is now if
users are confused. I bet a lot of our users don't have much or any
experience installing R packages that require more than just a call to
`install.packages()`.

Jonah

**Comment by jgabry** [comment]
I've started working on this. So far I have split the up the instructions so that Windows has its own page and Mac and Linux are together. 

https://github.com/stan-dev/rstan/wiki/RStan-Getting-Started

I also cleaned up some of the instructions, although they could still be a bit more checklisty.

**Comment by bob-carpenter** [comment]
Thanks!  That's already a big improvement breaking things out and including screen shots.  

A checklist would be super useful for all the reasons Michael outlined in the meeting.  What happens is that users get far down the road without verifying earlier steps. It's hard to keep track of where you're at with a wall of text, especially if you're impatient and/or inexperienced at installs requiring more than install.packages("foo").
